src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/TasksController.kt

```diff
@@ -58,15 +58,25 @@ class TasksController(
 ) : TasksApiDelegate {
   private val log = LoggerFactory.getLogger(this::class.java)
 
-  override fun tasksReallocatableGet(): ResponseEntity<List<Task>> {
+  override fun tasksReallocatableGet(type: String?): ResponseEntity<List<Task>> {
     val user = userService.getUserForRequest()
-
     if (user.hasRole(UserRole.CAS1_WORKFLOW_MANAGER)) {
-      val assessmentTasks = getAssessmentTasks(assessmentService.getAllReallocatable(), user)
-      val placementRequestTasks = getPlacementRequestTasks(placementRequestService.getAllReallocatable(), user)
-      val placementApplicationTasks = getPlacementApplicationTasks(placementApplicationService.getAllReallocatable(), user)
+      if (type != null) {
+        val taskType = enumConverterFactory.getConverter(TaskType::class.java).convert(
+          type.kebabCaseToPascalCase(),
+        ) ?: throw NotFoundProblem(type, "TaskType")
+
+        when (taskType) {
+          TaskType.assessment -> return assessmentTasksResponse(user)
+          TaskType.placementRequest -> return placementRequestTasks(user)
+          TaskType.placementApplication -> return placementApplicationTasks(user)
+          else -> {
+            throw ForbiddenProblem()
```

Comment:
I feel like this should be a `BadRequestProblem` if the task type isn't a valid one.